generators/app/gulp-tasks/project.js

```diff
@@ -0,0 +1,113 @@
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+'use strict';
+
+const mergeStream = require('merge-stream');
+const gulp = require('gulp');
+
+const polymer = require('polymer-build');
+const PolymerProject = polymer.PolymerProject;
+const fork = polymer.forkStream;
+const addServiceWorker = polymer.addServiceWorker;
+
+const polymerJSON = require(global.config.json);
+const project = new PolymerProject(polymerJSON);
+
+// This is the heart of polymer-build, and exposes much of the
+// work that Polymer CLI usually does for you
+// There are tasks to split the source files and dependency files into
+// streams, and tasks to rejoin them and output service workers
+// You should not need to modify anything in this file
+// If you find that you can't accomplish something because of the way
+// this module is structured please file an issue
+// https://github.com/PolymerElements/generator-polymer-init-custom-build/issues
+
+// Returns a ReadableStream of all the source files
+// Source files are those in src/** as well as anything
+// added to the sourceGlobs property of polymer.json
+function splitSource() {
+  return project.sources().pipe(project.splitHtml());
+}
+
+// Returns a ReadableStream of all the dependency files
+// Dependency files are those in bower_components/**
+function splitDependencies() {
+  return project.dependencies().pipe(project.splitHtml());
+}
+
+// Returns a WriteableStream to rejoin all split files
+function rejoin() {
+  return project.rejoinHtml();
+}
+
+// Returns a function which accepts refernces to functions that generate
+// ReadableStreams. These ReadableStreams will then be merged, and used to
+// generate the bundled and unbundled versions of the site.
+function merge(source, dependencies) {
+  return function bundle() {
+    // merge the source and dependencies streams so we can analyze the project
+    const mergedFiles = mergeStream(source(), dependencies())
+      .pipe(project.analyzer);
+
+    return Promise.all([
+      new Promise((resolve) => {
+        // this fork will vulcanize the project
+        fork(mergedFiles)
+          .pipe(project.bundler)
+          // write to the bundled folder
+          .pipe(gulp.dest(global.config.bundled))
+          .on('end', resolve);
+      }),
+      new Promise((resolve) => {
+        fork(mergedFiles)
+          // write to the unbundled folder
+          .pipe(gulp.dest(global.config.unbundled))
+          .on('end', resolve);
+      })
+    ]);
+  }
+}
+
+// Generates a service worker for the bundled and unbundled version of the site
+// using the settings from the global.config
+function serviceWorker() {
+  const swConfig = {
+    navigateFallback: global.config.fallback,
+  };
```

Comment:
@robdodson @chuckh @FredKSchott @addyosmani maybe we need to get this info from the `polymer.json` or from the `sw-precache-config.js`.